pytorch_lightning/utilities/parsing.py

```diff
@@ -254,13 +265,19 @@ def lightning_getattr(model, attribute):
 
 
 def lightning_setattr(model, attribute, value):
-    """ Special setattr for lightning. Checks for attribute in model namespace
-        and the old hparams namespace/dict.
-        Will also set the attribute on datamodule, if it exists.
     """
-    holders = lightning_get_all_attr_holders(model, attribute)
+    Special setattr for lightning. Checks for attribute in model namespace
+    and the old hparams namespace/dict.
+    Will also set the attribute on datamodule, if it exists.
+
+    Raises:
+        AttributeError:
+            If ``model`` doesn't have ``attribute`` in any of
+            model namespace, the hparams namespace/dict, and the datamodule.
```

Comment:
+1 for this. shall we add this in other places too. would be good to have the possible exceptions in the docs visible to the user.

Reply:
The tracking issue is here: #5434

Reply:
nicE!